src/x509.rs

```diff
@@ -246,6 +250,54 @@ impl OwnedX509Store {
     pub fn pointer(&self) -> *mut X509_STORE {
         self.raw
     }
+
+    pub fn len(&self) -> usize {
+        match unsafe { OPENSSL_sk_num(X509_STORE_get0_objects(self.raw) as *const OPENSSL_STACK) } {
+            -1 | 0 => 0,
+            i => i as usize,
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    pub fn add(
+        &mut self,
+        cert_ders: impl IntoIterator<Item = CertificateDer<'static>>,
+    ) -> Result<(), Error> {
+        for cert_der in cert_ders {
+            let item = OwnedX509::parse_der(cert_der.as_ref())
+                .ok_or_else(|| Error::bad_data("cannot parse certificate"))?;
+            match unsafe { X509_STORE_add_cert(self.raw, item.borrow_ref()) } {
+                1 => {}
+                _ => {
+                    return Err(Error::bad_data("cannot X509_STORE_add_cert"));
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn root_store(&self) -> Result<Arc<RootCertStore>, rustls::Error> {
```

Comment:
The `root_store()` name evokes a getter. Maybe `to_root_store()`? Was wondering about some kind of `TryFrom` or `TryInto` implementation but I think that gets a little ugly wrt coherence.